### ISSUE TITLE
GraphMachine fails with State object

### ISSUE THREAD

**Issue by mathiasimmer** [tyarkoni/transitions#215]
Consider the following piece of code:
```
from transitions import Machine
from transitions import State


class Matter(object):
    pass


lump = Matter()

solid = State('solid')
liquid = State('liquid')
gas = State('gas')
machine = Machine(lump, [solid, liquid, gas], initial=solid)

print lump.state
```
which correctly returns "solid".
However changing the import to
```
from transitions.extensions import GraphMachine as Machine
```
results in a failure:
```
(...) line 15, in <module> machine = Machine(lump, [solid, liquid, gas], initial=solid)
  File "/usr/local/lib/python2.7/dist-packages/transitions/extensions/diagrams.py", line 268, in __init__
    self.set_node_state(model.graph, self.initial, 'active')
  File "/usr/local/lib/python2.7/dist-packages/transitions/extensions/diagrams.py", line 323, in set_node_state
    node = _get_subgraph(node, 'cluster_' + node_name)
TypeError: cannot concatenate 'str' and 'State' objects
```

I suppose changing 
```
node = _get_subgraph(node, 'cluster_' + node_name)
```
to
```
node = _get_subgraph(node, 'cluster_' + str(node_name))
```
or
```
node = _get_subgraph(node, 'cluster_%s' % node_name)
```
will fix it.

**Comment by aleneum** [comment]
Hi @mathiasimmer,

the keyword `initial` currently expects a string. Even though the output of the lump.state is correct, the way the machine has been initialised is not.

Have a look at this constructed case, even though no one will probably initialise a machine like that:
```python

from transitions import Machine
from transitions import State


class Matter(object):
    pass


lump = Matter()

solid = State('solid')
second_solid = State('solid')
liquid = State('liquid')
gas = State('gas')
machine = Machine(lump, [solid, liquid, gas], initial=second_solid)

print machine.states['solid'] == second_solid # will return True
```
For `initial`, transitions expects a string and therefore just checks if a state is already named like the value of `initial`. If you pass a `State`, this test obviously fails. Consequently, `Machine.add_state` will add the passed `State` instance as a new state and override our previously added solid state.
You see that our first instance of solid has been kicked from the machine's states in favour of the second instance passed to initial.

Long story short: Passing a `State` object to `initial` has not been supported (yet) and the fact that is *somehow works* for `Machine` is not intentional.

The question is: Should transitions support `State` as a valid type for `initial`? I'd say it adds convenience. But let's wait for feedback.

**Comment by aleneum** [comment]
Hi again, 

according to #153, transitions should be able to process State instances passed to `initial` correctly. So this qualifies clearly as a bug. I fixed it and extended testing. I checked your provided code sample and it does not raise `TypeError` anymore.

Thank you for reporting this!
